Source/JavaScriptCore/parser/Lexer.cpp

```diff
@@ -1208,17 +1208,59 @@ template <bool shouldBuildStrings> ALWAYS_INLINE typename Lexer<T>::StringParseR
 
     const T* stringStart = currentSourcePtr();
 
+    using UnsignedType = std::make_unsigned_t<T>;
+    auto quoteMask = SIMD::splat<UnsignedType>(stringQuoteCharacter);
+    constexpr auto escapeMask = SIMD::splat<UnsignedType>('\\');
+    constexpr auto controlMask = SIMD::splat<UnsignedType>(0xE);
+    constexpr auto nonLatin1Mask = SIMD::splat<UnsignedType>(0xff);
+    auto vectorMatch = [&](auto input) ALWAYS_INLINE_LAMBDA {
+        auto quotes = SIMD::equal(input, quoteMask);
+        auto escapes = SIMD::equal(input, escapeMask);
+        auto controls = SIMD::lessThan(input, controlMask);
+        if constexpr (sizeof(T) == sizeof(LChar) || !shouldBuildStrings) {
```

Comment:
Nit: I would make this `std::is_same_v<T, LChar>` instead of `sizeof(T) == sizeof(LChar)`